src/api/procedures/utils.ts

```diff
@@ -556,6 +529,35 @@ function assertIsAccount(target: Signer): asserts target is Account {
   }
 }
 
+/**
+ * @hidden
+ *
+ * Asserts valid request for becoming a Primary Account (JoinIdentity and RotatePrimaryKeyToSecondary)
+ */
+async function assertJoinIdentityAuthorizationValid(
```

Comment:
If we consider name change this could be `assertAddOrRotateAccountAuthorizationValid` which sounds more clear to me. IDK maybe it sounds good only in my head 😄